client/my-sites/pages/virtual-page/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { CompactCard, Gridicon } from '@automattic/components';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { ExternalLink } from '@wordpress/components';
+import { decodeEntities } from '@wordpress/html-entities';
+import { useTranslate } from 'i18n-calypso';
+import { useDispatch } from 'react-redux';
+import EllipsisMenu from 'calypso/components/ellipsis-menu';
+import InfoPopover from 'calypso/components/info-popover';
+import PopoverMenuItem from 'calypso/components/popover-menu/item';
+import PopoverMenuItemClipboard from 'calypso/components/popover-menu/item-clipboard';
+import { addQueryArgs } from 'calypso/lib/route';
+import { infoNotice } from 'calypso/state/notices/actions';
+import { setLayoutFocus } from 'calypso/state/ui/layout-focus/actions';
+import { setPreviewUrl } from 'calypso/state/ui/preview/actions';
+import { recordEvent } from '../helpers';
+import { PageCardInfoBadge } from '../page-card-info';
+import type { SiteDetails } from '@automattic/data-stores';
+
+interface Props {
+	site: SiteDetails;
+	id: string;
+	type: string;
+	title: string;
+	description?: string;
+	previewUrl?: string;
+	isHomepage?: boolean;
+}
+
+const VirtualPage = ( { site, id, type, title, description, previewUrl, isHomepage }: Props ) => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const decodedTitle = decodeEntities( title );
+	const editorUrl = addQueryArgs(
+		{ templateId: id, templateType: type },
+		`/site-editor/${ site.slug }`
+	);
+
+	const handleMenuToggle = ( isVisible: boolean ) => {
+		if ( isVisible ) {
+			dispatch( recordEvent( 'Clicked More Options Menu' ) );
+		}
+	};
+
+	const recordEditPage = () => dispatch( recordEvent( 'Clicked Edit Page' ) );
+
+	const recordViewPage = () => dispatch( recordEvent( 'Clicked View Page' ) );
+
+	const viewPage = () => {
+		recordViewPage();
+		dispatch( setPreviewUrl( previewUrl ) );
+		dispatch( setLayoutFocus( 'preview' ) );
+	};
+
+	const copyPageLink = () => {
+		dispatch(
+			infoNotice( translate( 'Link copied to clipboard.' ), {
+				duration: 3000,
+			} )
+		);
+		dispatch( recordEvent( 'Clicked Copy Page Link' ) );
+	};
+
+	return (
+		<CompactCard key={ id } className="page">
+			<div className="page__main">
+				<a
+					className="page__title"
+					href={ editorUrl }
+					title={ translate( 'Edit %(title)s', {
+						textOnly: true,
+						args: { title: decodedTitle },
+					} ) }
+				>
+					<span>{ decodedTitle }</span>
+					{ isHomepage && (
+						<InfoPopover position="right">
+							{ translate(
+								'The homepage of your site displays the Home template. {{learnMoreLink}}Learn more{{/learnMoreLink}}.',
```

Comment:
Wait this is not always the case. If the theme does not have the Home template, then Index template will be used.

Reply:
Yes, I considered saying "The homepage of your site displays the Front Page, Home, or Index template" but I thought it was a bit long and the user might feel more confused.

Do you have any suggestions?

Reply:
Can we just not show this notice at all? Maybe it will confuse the user more. cc: @autumnfjeld ?

Because the correct template will be shown anyway when the user clicks on that page (shown in the Site Editor).

Reply:
Or we can parse the template name based on the postId? E.g. `pub/blank-canvas-blocks//index`, we can split by `//` and get the `index` suffix as the template name.

Reply:
I assume we need to let users know this page is different from the others. Otherwise, when they click on "...", they will be confused why the actions are different.

Reply:
I've changed it to fetch the title of the template, so it will show the correct template name. The reason we cannot split the post id is we still need a translated name 😂

Reply:
Nice solution.